doc/reference/devicetree/api.rst

```diff
@@ -387,6 +387,8 @@ device.
        interprocess-communication (IPC)
    * - zephyr,itcm
      - Instruction Tightly Coupled Memory node on some Arm SoCs
+   * - zephyr,ocm
+     - On-chip memory node on Xilinx Zynq-7000 and ZynqMP SoCs
```

Comment:
Should this be more generalised?

The "on-chip memory" in this context refers to the SRAM on the microprocessor chip itself as opposed to external DRAMs (e.g. DDR), and there are many SoCs that have such a construct -- of course, they could be named somewhat differently, but adding a new name for such a construct per SoC does not exactly sound like a sane thing to do.

Reply:
That's basically what I was expecting - that something as generic as an OCM, under this particular name, should be a common thing among various ARM SoCs. But it seems like the Xilinx SoCs are actually the first SoCs to have this feature. The existing types ITCM and DTCM make assumptions on the purpose of the integrated memory, which the OCM does not. ST at least uses a different name for their on-chip memory, I haven't looked into whether a specific purpose is implied for their CCM. So, at the time being, there don't seem to be any SoC families to generalise the OCM to. In case the addition of SoCs made by vendors other than those already supported is pending, I can check if those come with an identical OCM.